.github/workflows/ci.yaml

```diff
@@ -69,3 +69,25 @@ jobs:
       # - name: Setup SSH if failure
       #   if: failure()
       #   uses: lhotari/action-upterm@v1
+
+  check-contracts-changes:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for changes
+        uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            contracts_dir: 
+              - 'contracts/v2.6.2/**'
+
+      - name: Install scarb
+        if: steps.changes.outputs.contracts_dir == 'true'
+        run: curl --proto '=https' --tlsv1.2 -sSf https://docs.swmansion.com/scarb/install.sh | sh
+
+      - name: Run scarb check
+        if: steps.changes.outputs.contracts_dir == 'true'
+        working-directory: contracts/v2.6.2
+        run: scarb check
```

Comment:
Hi, if you read [How to contribute](https://github.com/focustree/starknet.dart/blob/e1c62fa12420582b7bd09632df03a35f5ffc4f9a/docs/how-to-contribute.mdx) and other jobs in you will see that this project is using [melos](https://melos.invertase.dev/~melos-latest/getting-started) for several tasks.

To avoid different behavior between developer and CI, smart contract checks should be done with a melos command